src/stress_test.cpp

```diff
@@ -0,0 +1,86 @@
+#include "verifier.h"
+#include <sstream>
+#include <string>
+#include <fstream>
+#include <thread>
+
+std::vector<uint8_t> HexToBytes(const char *hex_proof) {
+    int len = strlen(hex_proof);
+    assert(len % 2 == 0);
+    std::vector<uint8_t> result;
+    for (int i = 0; i < len; i += 2)
+    {
+        int hex1 = hex_proof[i] >= 'a' ? (hex_proof[i] - 'a' + 10) : (hex_proof[i] - '0');
+        int hex2 = hex_proof[i + 1] >= 'a' ? (hex_proof[i + 1] - 'a' + 10) : (hex_proof[i + 1] - '0');
+        result.push_back(hex1 * 16 + hex2);
+    }
+    return result;
+}
+
+void doit(int thread)
+{
+    std::ifstream infile("/Users/bill/downloads/newvdfs.txt");
```

Comment:
you should probably check in this file too.
Doing the file I/O in the threads, rather than up-front I think might risk the CPU work to not necessarily overlap. It could be that the threads end up blocking each other on disk I/O and then do the CPU work while other threads are still reading from disk.
Maybe this loop takes long enough to have some overlap, but I would expect it to be more effective to do all disk I/O on the main thread, before starting the worker threads.

Reply:
the file i currently have is a gigabyte, but ill check in a small one to demo. i just wanted something simple for the test so i had the threads each read the file. i am sure after being read once, it is buffered in memory anyway. much easier than having the master dole it out. I currently get around 700%+ CPU with 8 cores so the efficiency is pretty high.